sign.go

```diff
@@ -0,0 +1,36 @@
+package factom
+
+import (
+	"encoding/json"
+)
+
+type Signature struct {
+	PubKey    []byte `json:"pubkey"`
+	Signature []byte `json:"signature"`
+}
+
+func SignData(addr string, data []byte) (*Signature, error) {
+	params := &struct {
+		Address string `json:"address"`
```

Comment:
What are your thoughts on making the parameters more generic? Perhaps rather than passing in an address, they pass in the public key bytes? Or maybe a generic key string since a consumer of this endpoint won't always have the public key bytes for a factoid address?

Just thinking as we extend the functionality of walletd to support DIDs (and current identities, but those should get deprecated), there will be other key types that aren't strictly addresses, and this will be a very useful feature for such applications. If we can avoid an API change down the line, it might be worth considering here.

Reply:
> Or maybe a generic key string and we detect the type server side (since a consumer of this endpoint won't always have the public key bytes for a factoid address)?

yeah i think this is the way to go, any suggestions for naming "address" something more generic? "name"? "identifier"?